tests/test_cmd_organizations.py

```diff
@@ -45,29 +45,37 @@
 REGISTRY_DOM_NOT_FOUND_ERROR_ALT = "Error: bitergia.com not found in the registry"
 REGISTRY_EMPTY_OUTPUT = ""
 
-REGISTRY_OUTPUT = """Bitergia\tbitergia.net
-Bitergia\tbitergia.com *
+REGISTRY_OUTPUT = """Bitergia\tbitergia.com *
+Bitergia\tbitergia.net
 Example\texample.com
-Example\texample.org
 Example\texample.net
+Example\texample.org
 LibreSoft"""
 
 REGISTRY_OUTPUT_ALT = """Bitergia\tbitergia.net
-Example\texample.com
 Example\tbitergia.com
-Example\texample.org
+Example\texample.com
 Example\texample.net
+Example\texample.org
 LibreSoft"""
 
 REGISTRY_OUTPUT_EXAMPLE = """Example\texample.com
-Example\texample.org
 Example\texample.net
+Example\texample.org
 MyExample\tmyexample.com"""
 
 REGISTRY_OUTPUT_EXAMPLE_ALT = """Example\texample.com
 Example\texample.net"""
 
 
+def sort_lines(lines):
```

Comment:
Instead of doing this, what you have to do is sort the results in `orgs` command. Thus, users will have the same output, no matter which DBMS they are using.

Reply:
Good idea. I will have a look to learn how to do that.

Reply:
I've been looking at this. There are two chances:

* sorting in `cmd/organizations.py`. The code is:

```python
    def registry(self, term=None):
       ....
            orgs = api.registry(self.db, term)
            self.display('organizations.tmpl', organizations=orgs)
```

It would be adding `sorting` to orgs:

```python
    def registry(self, term=None):
       ....
            orgs = sorting(api.registry(self.db, term))
            self.display('organizations.tmpl', organizations=orgs)
```

Or instead, go to the real source, which is in `api.py`:

```
def registry(db, term=None):
    ...
        if term:
            orgs = session.query(Organization).\
                filter(Organization.name.like('%' + term + '%')).\
                order_by(Organization.name).all()
           ...
        else:
            orgs = session.query(Organization).\
                order_by(Organization.name).all()
        ...
    return orgs
```

Curiously enough, results (`orgs`) seem to be already ordered. Assuming that for some reason the ordering by MySQL is not the same as for MariaDB, the fix would be letting Python sort instead of the database:

```
def registry(db, term=None):
    ...
        if term:
            orgs = session.query(Organization).\
                filter(Organization.name.like('%' + term + '%')).\
                all()
           ...
        else:
            orgs = session.query(Organization).\
                all()
        ...
    return sorted(orgs)
```

But this seems to break other tests (maybe assuming a certain order in orgs?). I'm not sure if that's the reason or there is something else. Before proceeding, what do you think is more appropriate?

Reply:
Hmmm. Unfortunately it is not so easy. `orgs` in the code above (in both cases) are not strings, but Organization objects. That means they cannot be sorted. So, either I define a sorting method for them, or we forget about these solutions.

So, now I see three options:

* Leave it as it is, maybe with an accompanying ticket about finding out how to produce the same order with MySQL and MariaDB.
* Explore why both databases give a different order, and fix the code (instead of sorting in the test).
* Define ordering methods for Organization, and use them instead of sorted().

I would vote for the first one, since this was just trying to make the tests run, but I will try whatever you find convenient.

Reply:
Not completely sure, but maybe this has something to do...: [Why is ORDER BY in a FROM Subquery Ignored?](https://mariadb.com/kb/en/library/why-is-order-by-in-a-from-subquery-ignored/).

Reply:
I've been checking this. I think the problem is not in the way organizations are sorted but in the way their domains are sorted. I think the best way to fix it is to define how the relationship of `Organization` and `Domain` is sorted:  http://docs.sqlalchemy.org/en/latest/orm/extensions/orderinglist.html